### DISCUSSION TITLE
Two Factor Auth for Pocketbase Admin UI Dashboard

### DISCUSSION THREAD

**Discussion by jlwoolf** [pocketbase/pocketbase#1553] [Q&A]
I'd like to be able to keep the UI Dashboard accessible, but if possible, it would be nice to enable two factor auth for it (like a TOTP). Does anyone know how this might be possible. I am less experienced with backend dev, and currently self hosting pocketbase to learn more.

**Comment by ganigeorgiev** [comment]
There is no built-in two-factor authentication at the moment but a simple email-code verification (and maybe TOTP since other users also requested in https://github.com/pocketbase/pocketbase/discussions/1208) is in my todo and eventually will be implemented sometime in the future but for now it is a low priority.

If you want to restrict access to your Admin auth endpoint, you can attach to the [`app.OnAdminAuthRequest()` hook](https://pocketbase.io/docs/event-hooks/#admin-api-hooks) (_there will be also 2 additional auth hooks with the next release as part of https://github.com/pocketbase/pocketbase/issues/1240_) to validate your otp code and/or implement a [custom endpoint/middleware](https://pocketbase.io/docs/router/) to generate it.
